### PR TITLE
fix(EraserBrush): `intersectsWithObject` edge cases

### LINKED ISSUE #7277: Fabric.js Eraser not working properly if canvas is zoom in or zoom out
## Version
4.5.1

## Test Case
https://codesandbox.io/s/sparkling-dew-2e038

## Information about environment
Although I am using it in NextJS application. I made the test case in Vanilla JS
Which browsers?
Chrome, Firefox etc.

## Steps to reproduce
First, draw something on the board then zoom in a lot then try to Erase. 
## Expected Behavior
It Should erase the drawing.

https://user-images.githubusercontent.com/79884206/128333869-cf06abde-3bb5-4286-82f2-83f34df51d32.mp4

## Actual Behavior
it won't work. It mostly happening while I am trying to eraser around the corner of the drawing. It is also happening if I zoom out the canvas a lot.

### PR DESCRIPTION
**Pull request by ShaMan123**
closes #7277

### CODE REVIEW

File: src/mixins/eraser_brush.mixin.js [hunk 1]
```diff
@@ -929,6 +929,8 @@
         }
 
         var path = this.createPath(pathData);
+        //  needed for `intersectsWithObject`
+        path.setCoords();
```
Comment (asturur) [inline src/mixins/eraser_brush.mixin.js:933]: i m worried this won't work if there is some zoom level going on
But this for sure is fixing something, so for now is fine.
